### ISSUE TITLE
[HOW-TO] xrmApp.Entity.AddSubgridItem("xxx"); can't find element

### ISSUE THREAD

**Issue by issieeman** [Microsoft/EasyRepro#811]
Hi,
I'm having an issue where using the AddSubgridItem method returns the no such element error. 
When the application is fullscreen this does not work, because the button is hidden under the 3 dots. 
However if I make my screen smaller, the add new opportunity button does appear together with the refresh button and i can click them. 
anyone know how i can solve this? 

![afbeelding](https://user-images.githubusercontent.com/34887690/77704526-8a6b2080-6fbd-11ea-97c9-d8e0f4d8811c.png)

basically i need to click on new opportunity, but i can't find how to click on the 3 dots. 
by making my screen smaller, "new opportunity" isn't under the 3 dots and then i can use AddSubgridItem().

**Comment by jowells1** [comment]
@issieeman, can you try the following:
```csharp
xrmApp.RelatedGrid.ClickCommand("New Opportunity", null, true, "contactopportunitiesgrid");
```
Let's discuss the parameters being passed:
1. New Opportunity - This is the name of the button as you see it on the form
2. null - If the button led to a sub-area and another button, then you would pass the value of that label here
3. true - This is needed if you have to click the ellipses button (...)
4. contactopportunitiesgrid - This is the unique name of the sub-grid defined on your form.

In my test, I used the Contact entity doing a similar thing:
![image](https://user-images.githubusercontent.com/34693792/81135382-9089e080-8f1d-11ea-9cf2-d51b6cb1c309.png)

Let me know if you need more info or help.

Thanks,
Josh

**Comment by TYLEROL** [comment]
@issieeman - We have introduced some changes from the latest develop branch to releases/v9.1 and the 9.1.0.17138 NuGet package.

These changes include proper support for SubGrids. While the 'AddSubgridItem' method has been deprecated, you can instead use the following to click the appropriate command on the grid to create a new record:

`xrmApp.Entity.SubGrid.ClickCommand(string subGridSchemaName, string buttonName, string subButtonName);`

The 'RelatedGrid' methods will be used if accessing grids under the 'Related' tab section.

Warm Regards,
Tyler
